pkg/relay/ethereum/tx.go

```diff
@@ -338,3 +362,43 @@ func (c *Chain) SendTx(opts *bind.TransactOpts, msg sdk.Msg, skipUpdateClientCom
 	}
 	return tx, err
 }
+
+func (c *Chain) getRevertReasonFromReceipt(ctx context.Context, receipt *client.Receipt) (string, error) {
+	var errorData []byte
+	if receipt.HasRevertReason() {
+		errorData = receipt.RevertReason
+	} else if c.config.EnableDebugTrace {
+		callFrame, err := c.client.DebugTraceTransaction(ctx, receipt.TxHash)
+		if err != nil {
+			return "", err
+		} else if len(callFrame.Output) == 0 {
+			return "", fmt.Errorf("execution reverted without error data")
+		}
+		errorData = callFrame.Output
+	} else {
+		return "", fmt.Errorf("no way to get revert reason")
+	}
+
+	revertReason, err := c.errorRepository.ParseError(errorData)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse error: %v", err)
```

Comment:
@siburu For unknown error ABI, we should contain raw error data in an error.
For example, it is impossible to provide complete error ABIs for an App that allows external contract calls in the RecvPacket.

Reply:
@bluele done https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/39/commits/9a63ad9c7845ae65c40f7f7b2e792cf718c6ee95